src/mono/wasm/build/WasmApp.Native.targets

```diff
@@ -508,6 +508,8 @@ EMSCRIPTEN_KEEPALIVE void mono_wasm_load_profiler_aot (const char *desc) { mono_
       <_AotInputAssemblies Include="@(_WasmAssembliesInternal)">
         <AotArguments>@(MonoAOTCompilerDefaultAotArguments, ';')</AotArguments>
         <ProcessArguments>@(MonoAOTCompilerDefaultProcessArguments, ';')</ProcessArguments>
+        <!-- Skip known-huge assembly. This takes over ten minutes on a top end PC to AOT, and consumes over 8GiB RAM in the process. -->
+        <AOT_InternalForceToInterpret Condition="$([System.Text.RegularExpressions.Regex]::IsMatch('%(_WasmAssembliesInternal.FullPath)', '.+Microsoft\.CodeAnalysis\.CSharp\.dll')) ">true</AOT_InternalForceToInterpret>
```

Comment:
We can take a list of assembly names which should get skipped, which the user project can set. And here, we can set the metadata for the matching `@(_AotInputAssemblies)`.